### PR TITLE
Update documentation.

### PR DESCRIPTION
**Pull request by bierik**
Add documentation for showroom client API.

/cc @deiferni

### CODE REVIEW

**Comment by elioschmutz** [comment]
👍  finde ich gut so. @deiferni ist für dich auch i.O?

File: README.rst [hunk 1]
```diff
@@ -108,6 +122,186 @@ to enable source maps in Google Chrome, if you haven't already done so.
...
+--------------
+
+Creates an empty showroom instance with default options
+
+.. code:: javascript
+
+  let showroom = Showroom();
+
+The showroom constructor accepts a `NodeList <https://developer.mozilla.org/en/docs/Web/API/NodeList>`_ or a `jQuery <http://api.jquery.com/jquery/>`_.
+The items can provide a target as a HTML data attribute (**data-showroom-target**) to define the endpoint where the item will fetch its content from.
```
Comment (deiferni) [inline README.rst]: as a**n** HTML

File: README.rst [hunk 2]
```diff
@@ -108,6 +122,186 @@ to enable source maps in Google Chrome, if you haven't already done so.
...
+To make a title visible in the overlay provide a title as a HTML data attribute  (**data-showroom-title**).
```
Comment (deiferni) [inline README.rst]: same here

File: README.rst [hunk 3]
```diff
@@ -108,6 +122,186 @@ to enable source maps in Google Chrome, if you haven't already done so.
...
+
+.. code:: javascript
+
+  let items = $(".item");
+  let showroom = Showroom(items);
+
+Configuration
+-------------
+
+The showrooms provides the following options.
```
Comment (deiferni) [inline README.rst]: provide~~s~~

File: README.rst [hunk 4]
```diff
@@ -108,6 +122,186 @@ to enable source maps in Google Chrome, if you haven't already done so.
...
+
++----------+--------------------------+--------------------------------------------------------------------------+
+| Option   | Default                  | Description                                                              |
++==========+==========================+==========================================================================+
+| cssClass | "ftw-showroom"           | Class attribute on the root element of the gallery                       |
++----------+--------------------------+--------------------------------------------------------------------------+
+| render   | See rendering_. section  | Override the default render behavior                                     |
++----------+--------------------------+--------------------------------------------------------------------------+
+| tail     | Empty function           | Is getting called when the user reaches the last element of the gallery  |
```
Comment (deiferni) [inline README.rst]: Is getting called -> Called

File: README.rst [hunk 5]
```diff
@@ -108,6 +122,186 @@ to enable source maps in Google Chrome, if you haven't already done so.
...
++----------+--------------------------+--------------------------------------------------------------------------+
+| head     | Empty function           | Is getting called when the user reaches the first element of the gallery |
++----------+--------------------------+--------------------------------------------------------------------------+
+| fetch    | See fetching_. section   | Override the default fetch behavior                                      |
++----------+--------------------------+--------------------------------------------------------------------------+
+| template | See template section     | Override the default gallery template                                    |
++----------+--------------------------+--------------------------------------------------------------------------+
+| target   | body Element             | Define a selector where the gallery is getting to be attached            |
```
Comment (deiferni) [inline README.rst]: is getting to -> will

File: README.rst [hunk 6]
```diff
@@ -108,6 +122,186 @@ to enable source maps in Google Chrome, if you haven't already done so.
...
++----------+--------------------------+--------------------------------------------------------------------------+
+
+.. _fetching:
+
+Fetching
+--------
+
+The default fetching function makes an AJAX call with the provided target on the item
```
Comment (deiferni) [inline README.rst]: hmm this confuses me, plz rephrase, maybe:

The default fetching function uses the target provided by each item to make an AJAX call to retrieve its content.

?

File: README.rst [hunk 7]
```diff
@@ -108,6 +122,186 @@ to enable source maps in Google Chrome, if you haven't already done so.
...
+    }
+  );
+
+
+.. _rendering:
+
+Rendering
+---------
+
+The default rendering function returns a HTML string using the default `Handlebars <http://handlebarsjs.com/>`_ template
```
Comment (deiferni) [inline README.rst]: a**n** HTML

File: README.rst [hunk 8]
```diff
@@ -108,6 +122,186 @@ to enable source maps in Google Chrome, if you haven't already done so.
...
+
+.. code:: javascript
+
+  function render(content) {
+    return $.when(content).pipe((content) => {
+      return $(template({ showroom: data, content: content, item: register.current }));
+    });
+  }
+
+This function can be overridden in the options to change the default behavior.
```
Comment (deiferni) [inline README.rst]: This function can be overridden, see *[add Link to configuration]*

File: README.rst [hunk 9]
```diff
@@ -108,6 +122,186 @@ to enable source maps in Google Chrome, if you haven't already done so.
...
+--------
+
+The default fetching function makes an AJAX call with the provided target on the item
+to retrieve its content.
+
+.. code:: javascript
+
+  function fetch(item) { return $.get(item.target); };
+
+This function can be overridden in the options to change the default behavior.
```
Comment (deiferni) [inline README.rst]: This function can be overridden, see *[add link to configuration]*.

**Comment by deiferni** [comment]
nice docs update! 💛 ✨

**Comment by bierik** [comment]
@deiferni updated.. thanks for your help

**Comment by deiferni** [comment]
👍

**Pull request merged**